Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebsiteData.cpp

```diff
@@ -863,6 +863,45 @@ g_assert_nonnull(data);
     g_assert_null(dataList);
 }
 
+static void testWebViewHandleCorruptedLocalStorage(WebsiteDataTest* test, gconstpointer)
+{
+    const char html[] = "<html><script>let foo = (window.localStorage.length ? window.localStorage.getItem('item'):''); window.localStorage.setItem('item','value');</script></html>";
+    auto waitForFooChanged = [&test](WebKitWebView* webView) {
+        char* fooValue;
+            GUniqueOutPtr<GError> error;
+            JSCValue* jscvalue;
+            WebKitJavascriptResult* result = test->runJavaScriptAndWaitUntilFinished("foo;", &error.outPtr(), webView);
+            g_assert_no_error(error.get());
+            jscvalue = webkit_javascript_result_get_js_value(result);
+            fooValue = jsc_value_to_string(jscvalue);
+        return fooValue;
```

Comment:
This should return `adoptGRef(fooValue)` so you don't have to manually `g_free` the result.